rekolekcje-engine/src/verify/java/pl/oaza/warszawa/dor/rekolekcje/api/participants/utils/ParticipantsApiExpectations.java

```diff
@@ -40,70 +33,28 @@ public void okStatusIsReturned(ResultActions response) throws Exception {
-    assertThat(parsedJson.read("$.personalData.motherName", String.class)).isEqualTo(expectedParticipant.getMotherName());
-    assertThat(parsedJson.read("$.personalData.fatherName", String.class)).isEqualTo(expectedParticipant.getFatherName());
-    assertThat(parsedJson.read("$.personalData.christeningPlace", String.class)).isEqualTo(expectedParticipant.getChristeningPlace());
-    compareDates(parsedJson.read("$.personalData.christeningDate", String.class), expectedParticipant.getChristeningDate());
-    assertThat(parsedJson.read("$.personalData.emergencyContactName", String.class)).isEqualTo(expectedParticipant.getCloseRelativeName());
-    assertThat(parsedJson.read("$.personalData.emergencyContactNumber", Long.class)).isEqualTo(expectedParticipant.getCloseRelativeNumber());
-  }
-
-  private void verifyAddressValueInResponse(DocumentContext parsedJson, ParticipantData expectedParticipant) {
-    assertThat(parsedJson.read("$.address.streetName", String.class)).isEqualTo(expectedParticipant.getStreet());
-    assertThat(parsedJson.read("$.address.streetNumber", Integer.class)).isEqualTo(expectedParticipant.getStreetNumber());
-    assertThat(parsedJson.read("$.address.flatNumber", Integer.class)).isEqualTo(expectedParticipant.getFlatNumber());
-    assertThat(parsedJson.read("$.address.postalCode", String.class)).isEqualTo(expectedParticipant.getPostalCode());
-    assertThat(parsedJson.read("$.address.city", String.class)).isEqualTo(expectedParticipant.getCity());
-    assertThat(parsedJson.read("$.healthReport.currentTreatment", String.class)).isEqualTo(expectedParticipant.getCurrentTreatment());
-  }
-
-  private void verifyExperienceValueInResponse(DocumentContext parsedJson, ParticipantData expectedParticipant) {
-    assertThat(parsedJson.read("$.experience.kwcStatus", String.class)).isEqualTo(expectedParticipant.getKwcStatus());
-    compareDates(parsedJson.read("$.experience.kwcSince", String.class), expectedParticipant.getKwcSince());
-    assertThat(parsedJson.read("$.experience.numberOfCommunionDays", Integer.class)).isEqualTo(expectedParticipant.getNumberOfCommunionDays());
-    assertThat(parsedJson.read("$.experience.numberOfPrayerRetreats", Integer.class)).isEqualTo(expectedParticipant.getNumberOfPrayerRetreats());
-  }
-
-  @Hack("too tired to think...")
-  @WTF("null, wuut?")
-  @Facepalm("junior's code")
-  private void compareDates(String dateFromJson, LocalDateTime storedDate) {
-    if (dateFromJson == null && storedDate == null) {
-      return;
-    }
-
-    final ZonedDateTime parsedDateFromJson = dateFromJson != null ? ZonedDateTime.parse(dateFromJson) : null;
-    assertThat(parsedDateFromJson).isEqualTo(convertToUtcDateTime(storedDate));
-  }
-
-  private ZonedDateTime convertToUtcDateTime(LocalDateTime localDateTime) {
-    return localDateTime == null ? null : ZonedDateTime.of(localDateTime, ZoneOffset.UTC);
+    final Long parsedId = parsedJson.read("$.id", Long.class);
```

Comment:
odważnie :D

Reply:
no właśnie dzięki nowemu podejściu można to wszystko zaorać. Zostaje metoda, która ręcznie sprawdza, czy id jest nadane, a tak to wszystko się odbywa na poziomie porównywania całych jsonów. Stąd te problemy z doklejaniem id. Coś za coś. Teraz więcej testów operuje bezpośrednio na API, tzn testujemy sobie w izolacji dodawanie, a w innych testach korzystamy z tego dodawania zakładając, że działa. Jeśli przestanie, to wywali się na to test, i przez to nie zmylą nas false positivy. Tak samo z pobieraniem danych, przy testach pobierania wykorzystujemy niskopoziomowe operacje, żeby przetestować w izolacji tę funkcjonalność, a w innych testach już możemy tego używać. Dzięki temu mamy dużo większą elastyczność i ogranicza to konieczność utrzymania dużej ilości logiki niskopoziomowej, jak to miało miejsce do tej pory, zwłaszcza w klasie `ParticipantDatabase`.